bazel/foreign_cc/yaml_cpp.patch

```diff
@@ -0,0 +1,33 @@
+diff --git a/BUILD.bazel b/BUILD.bazel
+index b0b9016..e0ebfd9 100644
+--- a/BUILD.bazel
++++ b/BUILD.bazel
+@@ -1,3 +1,9 @@
++yaml_cpp_defines = select({
++    # On Windows, ensure static linking is used.
++    "@envoy//bazel:windows_x86_64": ["YAML_CPP_STATIC_DEFINE", "YAML_CPP_NO_CONTRIB"],
++    "//conditions:default": [],
++})
++
+ cc_library(
+     name = "yaml-cpp_internal",
+     visibility = ["//:__subpackages__"],
+@@ -11,4 +17,5 @@ cc_library(
+     includes = ["include"],
+     hdrs = glob(["include/**/*.h"]),
+     srcs = glob(["src/**/*.cpp", "src/**/*.h"]),
++    defines = yaml_cpp_defines,
```

Comment:
Could this be in copts instead so these don't propagate up?

Reply:
They unfortunately need to propagate up or build on windows fails due to the API of yaml_cpp. I tried the copts in the beginning, but never to succeeded.